frontend/src/metabase/admin/routes.jsx

```diff
@@ -54,103 +59,130 @@ import GroupDetailApp from "metabase/admin/people/containers/GroupDetailApp";
 // Permissions
 import getAdminPermissionsRoutes from "metabase/admin/permissions/routes";
 
-const getRoutes = (store, IsAdmin) => (
-  <Route path="/admin" component={withBackground("bg-white")(IsAdmin)}>
+const createRouteGuard = userPredicate => {
+  const Wrapper = UserAuthWrapper({
+    predicate: currentUser =>
+      currentUser?.is_superuser || userPredicate(currentUser),
+    failureRedirectPath: "/unauthorized",
+    authSelector: state => state.currentUser,
+    allowRedirectBack: false,
+    wrapperDisplayName: "RouteGuard",
+    redirectAction: routerActions.replace,
+  });
+
+  return Wrapper(({ children }) => children);
+};
+
+const DataModelGuard = createRouteGuard(
```

Comment:
Would it make sense to define this `DataModelGuard` in main `routes.jsx` file and pass it like `CanAccessSettings` as well?